File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/reactiveobject/BrowseObjectRequest.java
package org.opcfoundation.webserver.service.message.reactiveobject;

import org.eclipse.milo.opcua.stack.core.types.structured.BrowseDescription;
import org.opcfoundation.webserver.types.common.UaBrowseAdditionalInfo;
import org.opcfoundation.webserver.types.common.UaObjectId;

public class BrowseObjectRequest {
    private final UaObjectId objectId;
    private final UaBrowseAdditionalInfo additionalInfo;
    private final BrowseDescription browseDescription;

    public BrowseObjectRequest(
            UaObjectId objectId,
            UaBrowseAdditionalInfo additionalInfo,
            BrowseDescription browseDescription)
    {
        this.objectId = objectId;
        this.additionalInfo = additionalInfo;
        this.browseDescription = browseDescription;
    }

    public UaObjectId getObjectId() {
        return objectId;
    }

    public UaBrowseAdditionalInfo getAdditionalInfo() {
        return additionalInfo;
    }

    public BrowseDescription getBrowseDescription() { return browseDescription; }
}

File: opcua-webservice-node/src/server/service/message/reactiveobject/BrowseObjectRequest.ts
import { UaBrowseDescription } from "opcua-webapi-ts";
import { UaBrowseAdditionalInfo, UaObjectId } from "../../../types";

export class BrowseObjectRequest {
    private readonly _objectId: UaObjectId;
    private readonly _additionalInfo: UaBrowseAdditionalInfo;
    private readonly _browseDescription: UaBrowseDescription;

    constructor(
        objectId: UaObjectId,
        additionalInfo: UaBrowseAdditionalInfo,
        browseDescription: UaBrowseDescription,
    ) {
        this._objectId = objectId;
        this._additionalInfo = additionalInfo;
        this._browseDescription = browseDescription;
    }

    get objectId(): UaObjectId {
        return this._objectId;
    }

    get additionalInfo(): UaBrowseAdditionalInfo {
        return this._additionalInfo;
    }

    get browseDescription(): UaBrowseDescription {
        return this._browseDescription;
    }
}